src/agni/server/dataguard/I_UserDataGuard.java

```diff
@@ -19,4 +19,16 @@
     public void changeUserCurrentChat(String user, String chatname);
 
     public String userCurrentChat(String user);
+
+	public boolean groupExists(String groupName);
+
+	public String getUsername(String ip);
+
+	public void addUserToChat(String username, String groupName);
+
+	public void removeUserFromChat(String username, String groupName);
```

Comment:
Wasn't that what I was trying to move around yesterday but then we decided to put it back? Because I thought the GroupChatDataGuard pertained to a single group chat?